Add np.linalg.norm implementation

arjavtrivedi opened the pull request:
- [x] Closes #1250 
- [x] Executed ``pre-commit run --all-files`` with no errors
- [x] The change is fully covered by automated unit tests
- [x] Documented in docs/ as appropriate
- [x] Added an entry to the CHANGES file

Doc entry in _docs/numpy.ipynb_ already covered by print statement here:
```python
from pint.numpy_func import HANDLED_FUNCTIONS
print(sorted(list(HANDLED_FUNCTIONS)))
```

keewis commented on pint/testsuite/test_issues.py:
```diff
@@ -817,6 +817,12 @@ def test_issue_1185(self):
         )
 
+    @helpers.requires_numpy
+    def test_issue_1250(self):
```
I would put this in `test_numpy.py` and call it `test_linalg_norm`, but otherwise this looks good to me.

arjavtrivedi replied:
@keewis - thanks for the feedback. I've removed this test in favour of the implementation in `test_numpy.py` ([here](https://github.com/hgrecco/pint/blob/cc56b12ca5e07ea756a71847d30635f370415344/pint/testsuite/test_numpy.py#L1429)).

keewis reviewed (approved):
looks good to me, but someone with the permissions would have to rerun the tests so we can investigate the failures (the logs have expired since then).

arjavtrivedi commented:
I've managed to come back to this PR after quite a while. As such, I've updated merged upstream changes and addressed the previous PR feedback. The actions on the forked repository ([here](https://github.com/arjavtrivedi/pint/actions)) appear to be passing, but otherwise not sure why the PR workflows here are failing.

AdVetter commented:
We are using Pint and `np.linalg.norm` a lot in our calculations and had to write helper a function to use it properly. Looking really forward to when this is merged! :-)

keewis commented:
@arjavtrivedi (or @hgrecco / @jules-ch), could you merge in `master`? That should get RTD to build, and we'd need a maintainer to once again approve the workflow runs. Once tests pass I think this should be ready.

arjavtrivedi commented:
Thank you @hgrecco for the great software!

keewis commented:
> unfortunately I don't have write access else I'd merge!

This is not important anymore since the PR was merged (but I guess is good to know for any future PRs). Still, what I meant was to merge `master` into *this* branch to make use of some changes in `master` that fixed RTD.

hgrecco commented:
@keewis I think that the current master has everything ok, right?
